packages/plugin-core/src/utils/pods.ts

```diff
@@ -17,3 +26,90 @@ export const showPodQuickPickItemsV4 = (podItem: PodItemV4[]) => {
     canPickMany: false,
   });
 };
+
+export const getOauthClient = async () => {
+  const port = fs.readFileSync(
+    path.join(DendronWorkspace.wsRoot(), ".dendron.port"),
+    { encoding: "utf8" }
+  );
+
+  const stringifiedParams = queryString.stringify({
+    client_id:
+      "587163973906-od2u5uaop9b2u6ero5ltl342hh38frth.apps.googleusercontent.com",
+    redirect_uri: `http://localhost:${port}/api/oauth/getToken`,
+    scope: gdocScope.join(" "), // space seperated string
+    response_type: "code",
+    access_type: "offline",
+    prompt: "consent",
+  });
+  await open(
+    `https://accounts.google.com/o/oauth2/v2/auth?${stringifiedParams}`
+  );
+};
+export const showDocumentQuickPick = async (
+  docs: string[]
+): Promise<{ label: string } | undefined> => {
+  /** Least Recently Used Documents */
+  let LRUDocs: string[] | undefined =
+    await DendronWorkspace.instance().getGlobalState(GLOBAL_STATE.LRUDocs);
+  LRUDocs = _.isUndefined(LRUDocs) ? [] : LRUDocs;
+
+  docs = docs.filter((doc) => !LRUDocs?.includes(doc));
+
+  const pickItems = LRUDocs.concat(docs).map((doc) => {
+    return {
+      label: doc,
+    };
+  });
+
+  return window.showQuickPick(pickItems, {
+    placeHolder: "Choose a document",
+    ignoreFocusOut: false,
+    matchOnDescription: true,
+    canPickMany: false,
+  });
+};
+
+export const getHierarchyDest = async (
+  title: string
+): Promise<string | undefined> => {
+  const hierarchyDestination = await window.showInputBox({
```

Comment:
actually what I had in mind around what to store in global state was the destination for a particular note.  So if previously we had put the `meet.metrics` gdoc into the `dendron.meet.metrics` hierarchy, then we remember that and autopopulate it in `placeHolder`.   We can add this in v2.

Reply:
This is what we aimed for in this v1 as well. In the loom video, when we try to import again the same doc `meet.planning`, The placeholder is auto populated with `user.joshi.planning` since it was imported here in the first attempt.

Reply:
O - I didn't see the Loom video at first and misread the code haha.  Great!